roles/openshift_logging/tasks/set_defaults_from_current.yml

```diff
@@ -11,24 +11,236 @@
+
+    - conditional_set_fact:
+        facts: "{{ openshift_logging_facts['curator']['deploymentconfigs']['logging-curator']['containers'] | flatten_dict }}"
+        vars:
+          __openshift_logging_curator_cpu_limit: curator.resources.limits.cpu
+          __openshift_logging_curator_memory_limit: curator.resources.limits.memory
+          __openshift_logging_curator_cpu_request: curator.resources.requests.cpu
+
+
+# Curator Ops
+- when: openshift_logging_facts['curator_ops']['deploymentconfigs']['logging-curator-ops'] is defined
+  block:
+    - set_fact:
+        __openshift_logging_curator_ops_default_days: "{{ openshift_logging_facts['curator_ops']['deploymentconfigs']['logging-curator-ops']['containers']['curator']['env'] | entry_from_name_value_pair('CURATOR_DEFAULT_DAYS') }}"
+        __openshift_logging_curator_ops_run_hour: "{{ openshift_logging_facts['curator_ops']['deploymentconfigs']['logging-curator-ops']['containers']['curator']['env'] | entry_from_name_value_pair('CURATOR_RUN_HOUR') }}"
+        __openshift_logging_curator_ops_run_minute: "{{ openshift_logging_facts['curator_ops']['deploymentconfigs']['logging-curator-ops']['containers']['curator']['env'] | entry_from_name_value_pair('CURATOR_RUN_MINUTE') }}"
+        __openshift_logging_curator_ops_run_timezone: "{{ openshift_logging_facts['curator_ops']['deploymentconfigs']['logging-curator-ops']['containers']['curator']['env'] | entry_from_name_value_pair('CURATOR_RUN_TIMEZONE') }}"
+
+    - conditional_set_fact:
+        facts: "{{ openshift_logging_facts['curator_ops']['deploymentconfigs']['logging-curator-ops']['containers'] | flatten_dict }}"
+        vars:
+          __openshift_logging_curator_ops_cpu_limit: curator.resources.limits.cpu
+          __openshift_logging_curator_ops_memory_limit: curator.resources.limits.memory
+          __openshift_logging_curator_ops_cpu_request: curator.resources.requests.cpu
+
+
+# Fluentd
+- when: openshift_logging_facts['fluentd']['daemonsets']['logging-fluentd'] is defined
+  block:
+    - set_fact:
+        __openshift_logging_fluentd_file_buffer_limit: "{{ openshift_logging_facts['fluentd']['daemonsets']['logging-fluentd']['containers'][0]['env'] | entry_from_name_value_pair('FILE_BUFFER_LIMIT') }}"
+        __openshift_logging_fluentd_buffer_queue_limit: "{{ openshift_logging_facts['fluentd']['daemonsets']['logging-fluentd']['containers'][0]['env'] | entry_from_name_value_pair('BUFFER_QUEUE_LIMIT') }}"
+        __openshift_logging_fluentd_buffer_size_limit: "{{ openshift_logging_facts['fluentd']['daemonsets']['logging-fluentd']['containers'][0]['env'] | entry_from_name_value_pair('BUFFER_SIZE_LIMIT') }}"
+
+    - conditional_set_fact:
+        facts: "{{ openshift_logging_facts['fluentd']['daemonsets']['logging-fluentd']['containers'][0] | flatten_dict }}"
```

Comment:
Is containers an array or a dict?  Do we need to map the array to become a dict?  I'm thinking of changes that may causes issues in the future like if we have to add a metrics container to the fluent image

Reply:
I actually was just thinking about this. I'm going to update the daemonset fact collecting for containers so that it will be a dict in the logging facts afterwards. Otherwise yes, it is a list.

Reply:
Updated this